mcs/tools/mdoc/Mono.Documentation/monodocer.cs

```diff
@@ -2013,7 +2010,10 @@ private void UpdateExtensionMethods (XmlElement e, DocsNodeInfo info)
 		member.AppendChild (link);
 		AddTargets (em, info);
 
-		extensionMethods.Add (em);
+		var sig = em.SelectSingleNode ("Member/MemberSignature[@Language='C#']/@Value");
+		if (!multiassembly || (multiassembly && sig != null && !extensionMethods.Any (ex => ex.SelectSingleNode ("Member/MemberSignature[@Language='C#']/@Value").Value == sig.Value))) {
```

Comment:
I wonder if having a `multiassembly` field is really the best idea, in terms of future evolution. *Now*, `multiassembly` is implied when `-apistyle` is used. Will that *always* be the case? *Should it always be the case*?

What might be better is, instead of having two variables which are tied together, use one variable -- `apistyle` -- and a method to provide the logic:

    bool IsMutiAssembly {
        get { return apistyle == "classic" || apistyle == "unified"; }
    }

    void AddExtensionMethod (XmlNode em)
    {
        bool multiassembly = IsMultiAssembly;
        var sig = em.SelectSingleNode ("Member/MemberSignature[@Language='C#']/@Value");
        if (!multiassembly ||
                (multiassembly && sig != null &&
                 !extensionMethods.Any (ex => ex.SelectSingleNode ("Member/MemberSignature[@Language='C#']/@Value").Value == sig.Value))) {
            extensionMethods.Add (em);
        }
    }